Sources/Main.cpp

```diff
@@ -136,32 +85,158 @@ void TX584Form::LoadFile(UnicodeString FileName)
+        UnicodeString str = List->Strings[i + 2];
+        unsigned code = StrToInt(str.SubString(1, 3)),
+                 reg = StrToInt(str.SubString(5, 3)),
+                 carry = StrToInt(str.SubString(9, 1)),
+                 op = StrToInt(str.SubString(11, 3));
+        //формируем код микроинструкции
+        unsigned opcode = code < 54 ? iSet[ReCode[code]].BitValue : NOP;
+        if (reg != 0xFF)
+            opcode |= reg;
+        opcode |= carry ? ATTR_CARRY : 0;
+        if (op != 0xFF)
+            opcode |= op << 5;
+        //определяем, используется ли перенос в данной инструкции
+        if (code < 54)
+            opcode |= CPU.FindOperand(ReCode[code], OP_CARRY, opcode) ? ATTR_CUSED : 0;
+        //записываем микроинструкцию в редактор кода
+        Code[i] = opcode;
+        CPU.Format(opcode, buf);
+        CodeListView->Items->Item[i]->SubItems->Strings[1] = buf;
+
+        UnicodeString comment = str.SubString(15, str.Length() - 14);
+        int Dummy;
+        if (ParseComment(comment, Dummy)) {
+            CodeListView->Items->Item[i]->SubItems->Strings[2] = str.SubString(15, str.Length() - 14);
+            CodeListView->Items->Item[i]->SubItems->Strings[3] = L"";
+        } else {
+            CodeListView->Items->Item[i]->SubItems->Strings[3] = str.SubString(15, str.Length() - 14);
+            CodeListView->Items->Item[i]->SubItems->Strings[2] = L"";
+        }
+    }
+    //очищаем остальные строки
+    for (int i = List->Count - 2; i < MAX_ADDR; i++) {
+        Code[i] = NOP;
+        CodeListView->Items->Item[i]->SubItems->Strings[1] = NOP_TEXT;
+        CodeListView->Items->Item[i]->SubItems->Strings[2] = L"";
+        CodeListView->Items->Item[i]->SubItems->Strings[3] = L"";
+    }
+    delete List;
+}
 //---------------------------------------------------------------------------
 
 void TX584Form::SaveFile(UnicodeString FileName)
 {
     TFileStream *Stream;
+    TBinaryWriter *Writer;
     try  {
         Stream = new TFileStream(FileName, fmCreate);
+        Writer = new TBinaryWriter(Stream, TEncoding::UTF8, false);
+
         //сохраняем в родном формате
-        unsigned Sign = X584;
-        Stream->Write(&Sign, 4);
+        unsigned int Sign = X584;
+        Writer->Write(Sign);
         for (int i = 0; i < MAX_ADDR; i++) {
             //сохраняем инструкцию
             Stream->Write(&Code[i], 2);
             //сохраняем комментарий
-            AnsiStringT<1251> str = CodeListView->Items->Item[i]->SubItems->Strings[2];
+            AnsiStringT<1251> control = CodeListView->Items->Item[i]->SubItems->Strings[2];
+            AnsiStringT<1251> comment = CodeListView->Items->Item[i]->SubItems->Strings[3];
+            AnsiStringT<1251> str;
+
+            if (control.Length() > 0) {
+                str = control;
+            }
+            else {
+                str = comment;
+            }
+
+            if (str.Length() >= 255) {
+                str = str.SubString0(0, 255);
```

Comment:
Use `SetLength` method to truncate the string to 255 characters. It is more simple and intuitive.